users/tracing/users.go

```diff
@@ -22,81 +22,85 @@ const (
 var _ users.UserRepository = (*userRepositoryMiddleware)(nil)
 
 type userRepositoryMiddleware struct {
-	tracer opentracing.Tracer
-	repo   users.UserRepository
+	tracer     opentracing.Tracer
+	repo       users.UserRepository
+	instanceID string
 }
 
 // UserRepositoryMiddleware tracks request and their latency, and adds spans
 // to context.
-func UserRepositoryMiddleware(repo users.UserRepository, tracer opentracing.Tracer) users.UserRepository {
+func UserRepositoryMiddleware(repo users.UserRepository, tracer opentracing.Tracer, instanceID string) users.UserRepository {
 	return userRepositoryMiddleware{
-		tracer: tracer,
-		repo:   repo,
+		tracer:     tracer,
+		repo:       repo,
+		instanceID: instanceID,
 	}
 }
 
 func (urm userRepositoryMiddleware) Save(ctx context.Context, user users.User) (string, error) {
-	span := createSpan(ctx, urm.tracer, saveOp)
+	span := createSpan(ctx, urm.tracer, saveOp, urm.instanceID)
```

Comment:
No need to pass `urm.instanceID` everytime, you can embed the function `createSpan` to the struct `userRepositoryMiddleware`